app/src/common/shared/org/mozilla/vrbrowser/SessionStore.java

```diff
@@ -727,7 +714,7 @@ public void onLocationChange(GeckoSession aSession, String aUri) {
 
         // The homepage finishes loading after the region has been updated
         if (mRegion != null && aUri.equalsIgnoreCase(SessionStore.HOME_WITHOUT_REGION_ORIGIN)) {
-            aSession.loadUri(SessionStore.HOME_WITHOUT_REGION_ORIGIN + "#region=" + mRegion);
+            aSession.loadUri("javascript:window.location.replace('" + getHomeUri() + "');");
```

Comment:
curious about the thinking behind this.

this might cause unintended/confusing refreshes/redirects, but it may probably best we handle this on the server, so as to not confusing URLs where the `?region=<mRegion>` does not match the `#region=<mRegion>`.

I was thinking we could do this to avoid the refresh/redirect:

```java
aSession.loadUri("javascript:window.location.hash = (window.location.hash ? window.location.hash + '&region=' : '#') + " + mRegion + "');
```

thoughts?

Reply:
loadUri doesn't support `javascript:` URLs, that's why it was changed.

Reply:
I guess it isn't supposed to work with remote pages but if it is working I guess it's okay.

Reply:
@cvan Not sure if I get it. The problem with URL loading (being ? or #) is that although there is no server request they get commited to the session history. What's the drawback of the window.location.replace? I thought that might help refreshing locally without making server requests or adding history entries.

Using:
```
aSession.loadUri("javascript:window.location.hash = (window.location.hash ? window.location.hash + '&region=' : '#') + " + mRegion + "');
```

will add a history entry.